### PR TITLE
OCPEDGE-1344: Updated details of installation and unanswered questions.

### PR DESCRIPTION
**Pull request by jaypoulz**
In this update, I went over the installation section and provided firmer details around the installation strategy. I tried to highlight where there were questions yet to be answered. I also started fleshing out the HyperShift and Version Skew sections.

### CODE REVIEW

File: enhancements/two-nodes-openshift/2no.md [hunk 1]
```diff
@@ -119,12 +119,24 @@ When starting etcd, the OCF script will use etcd's cluster ID and version counte
 
 #### Cluster Creation
 
-User creation of a two-node control-plane will be possible via the Assisted Installer.
+User creation of a two-node control-plane will be possible via the Assisted Installer. A key requirement of the project is that it can be deployed using only 2 nodes, so deployments will take advantage of the Assisted Installer's ability to use one of the target machines as the bootstrap node before it is rebooted into a control-plane node. There is a critical transition during this process, where in order to maintain etcd quorum, the bootstrap node will need to be removed from the etcd cluster before it is rebooted so that quorum can be maintained as the machine reboots into a second control-plane.
```
Comment (mshitrit) [inline enhancements/two-nodes-openshift/2no.md]: Nit: might be worth explaining in couple of words the reasoning for the only 2 nodes requirment

File: enhancements/two-nodes-openshift/2no.md [hunk 2]
```diff
@@ -119,12 +119,24 @@ When starting etcd, the OCF script will use etcd's cluster ID and version counte
...
 
-The procedure follows the standard flow except for the configuration of 2 nodes instead of 3, and the collection of RedFish details (including passwords!) for each node which are needed for the RHEL-HA configuration.
-If available via the SaaS offering (not confirmed, ZTP may be the target), the offering will need to ensure passwords are appropriately handled. 
+Otherwise, the procedure follows the standard flow except for the configuration of 2 nodes instead of 3. At this time we've discussed the collection of RedFish details (including passwords!) for each node. This is needed for the RHEL-HA configuration by leveraging the BareMetalHost CRDs populated from the baremetal platform specification in the install-config. There are open questions on how to ensure that pacemaker is the only entity responsible for fencing to prevent conflicting requests from the baremetal operator.
```
Comment (mshitrit) [inline enhancements/two-nodes-openshift/2no.md]: Regarding the questions about pacemaker being the only fencing entity, do you refer to RHWA operators  ? (i.e NHC, SNR, FAR etc...)

File: enhancements/two-nodes-openshift/2no.md [hunk 3]
```diff
@@ -119,12 +119,24 @@ When starting etcd, the OCF script will use etcd's cluster ID and version counte
...
 
-Everything else about cluster creation will be an opaque implementation detail not exposed to the user. 
+An important facility of the installation flow is the transition from a CEO deployed etcd to one controlled by pacemaker. The basic transition works as follows:
+1. MCO Extensions are used to ensure that pacemaker, corosync, and resource agents are pre-configured on CoreOS using installation manifests.
+2. Upon detection that the cluster infrastructure is using the DualReplica controlPlaneTopology in the infrastructure config, an in-cluster entity (see open questions regarding whether this should be handled by CEO or an additional operator) will run a command on one of the cluster nodes to initialize pacemaker. The outcome of this is that the resource agent will be started on both nodes.
+3. The aforementioned in-cluster entity will set the CEO configuration for managing to etcd external. When this happens, CEO removes the etcd pod from the static pod configs. The resource agents for etcd are running from step 2, and they are configured to wait for etcd pods to be gone so they can restart them using Podman. This results in near zero downtime for etcd.
```
Comment (mshitrit) [inline enhancements/two-nodes-openshift/2no.md]: Nit: "CEO configuration for managing to etcd external" 
Personally I find this sentence a bit confusing in the sense that it's not completely clear to me that CEO is relinquishing the control of etcd management.

Comment (mshitrit) [inline enhancements/two-nodes-openshift/2no.md]: Nit: "This results in near zero downtime for etcd"
IIUC the impact here is on the duration of the installation time, which is minor in that resolution.
If that's indeed the case and that's the only impact I'd consider omitting this part.

File: enhancements/two-nodes-openshift/2no.md [hunk 4]
```diff
@@ -198,6 +210,8 @@ More information on creating OCF agents can be found in the upstream [developer
 
 Tools for extracting support information (must-gather tarballs) will be updated to gather relevant logs for triaging issues.
 
+As part of the fencing setup, process ownership for cri-o and kubelet will remain the same. The main difference is that the resource agent will be responsible for signaling systemd to make changes to their active states. The etcd pods are different in this respect since they will be restarted using Podman. It may be possible to start these with the same user account as the original pods.
```
Comment (mshitrit) [inline enhancements/two-nodes-openshift/2no.md]: TBH I'm a bit fuzzy on this part
@clobrano WDYT ?

File: enhancements/two-nodes-openshift/2no.md (hunk 2 above)
Comment (jaypoulz) [reply enhancements/two-nodes-openshift/2no.md]: I was making reference to BMO - which doesn't manage fencing but it does care about power management for nodes.

File: enhancements/two-nodes-openshift/2no.md (hunk 4 above)
Comment (clobrano) [reply enhancements/two-nodes-openshift/2no.md]: It is correct :+1:

File: enhancements/two-nodes-openshift/2no.md (hunk 2 above)
Comment (jaypoulz) [reply enhancements/two-nodes-openshift/2no.md]: I added a line to note the optional operators.

**Pull request merged**